streams/src/test/java/org/apache/kafka/streams/state/internals/ListValueStoreTest.java

```diff
@@ -177,27 +180,47 @@ public void shouldGetAllReturnTimestampOrderedRecords() {
     }
 
     @Test
-    public void shouldAllowDeleteWhileIterateRecords() {
-        listStore.put(0, "zero1");
-        listStore.put(0, "zero2");
-        listStore.put(1, "one");
+    public void shouldAllowDeleteWhileIterateRecords2() {
```

Comment:
Adding more randomness into this test to check if deleting after iterating could still have defined behavior.